core/transaction.go

```diff
@@ -171,6 +237,15 @@ type DeployAccountTransaction struct {
 	TransactionSignature []*felt.Felt
 	// The transaction nonce.
 	Nonce *felt.Felt
+
+	// Version 3 fields
+	// See InvokeTransaction for descriptions of the fields.
+	ResourceBounds        map[Resource]*ResourceBounds
+	Tip                   uint64
+	PaymasterData         []*felt.Felt
+	AccountDeploymentData []*felt.Felt
+	NonceDAMode           DataAvailabilityMode
+	FeeDAMode             DataAvailabilityMode
```

Comment:
Do you think it makes sense put all these in a separate struct and embed them in each txn type, instead of duplicating them?

Reply:
If it is backwards compatible (ie wont need a DB migration), we can put all common fields in that struct as well.

Reply:
Just realized the `AccountDeploymentData` field is unused in `DeployAccount` transactions. Fixed that.

> Do you think it makes sense put all these in a separate struct and embed them in each txn type, instead of duplicating them?

Most of the fields serve very different purposes. I'm hesitant to divide the structs. Makes the code harder to read imho.

Reply:
> Most of the fields serve very different purposes

Do fields introduced in V3 have different meanings depending on the txn type?

Reply:
No.